app/mixins/connection-type.js

```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+
+const {Mixin, inject} = Ember;
+
+export default Mixin.create({
+	fastboot: inject.service(),
+	connectionType: 'unresolved',
+
+	init() {
+		if (!this.get('fastboot.isFastBoot')) {
```

Comment:
What do you think about using a computed property instead of "init"?

E.g. connectionType would be a computed property that recalculates on 'fastBoot.isFastboot'.

What it gives us? No race conditions: if you fetch a computed property using a getter, you are sure the logic below was or will be executed, no matter where you use the mixin :)